src/toHaveStyleRule.js

```diff
@@ -73,9 +95,26 @@ const getDeclaration = (rule, property) =>
 const getDeclarations = (rules, property) =>
   rules.map(rule => getDeclaration(rule, property)).filter(Boolean)
 
+const getModifierName = (isEnzyme, component) =>
+  `.${getClassNames(
+    isEnzyme ? Enzyme.mount(component) : renderer.create(component).toJSON()
+  )[0]}`
+
+const normalizeModifier = (isEnzyme, modifiers = []) =>
+  modifiers
+    .map(
+      modifier =>
+        isStyledComponent(modifier)
+          ? getModifierName(isEnzyme, React.createElement(modifier))
```

Comment:
Looks like the CI tests are failing, probably because of this. I couldn't really figure out how else to get the rendered static class name from the component: some direction would be appreciated here.